storage/range.go

```diff
@@ -1035,6 +1058,47 @@ func (r *Range) handleSkippedIntents(args proto.Request, intents []proto.Intent)
 	}
 }
 
+type replicaCorruptionError struct {
+	error
+	msg       string
```

Comment:
this much state is unusual, right? can you add a comment explaining what it's all for?